### ISSUE TITLE
ENH-000157979: Epic: [autocomplete, autocomplete-item] improve screen reader behavior

### ISSUE THREAD

**Issue by DintaMel** [Esri/calcite-components#14878]
**monday.com sync:** #12642303240

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Autocomplete, Autocomplete Item

### Summary

Autocomplete and Autocomplete Item can provide a better experience for screen reader users by:

- Announcing the number of results from a current search.
- Reading aloud Autocomplete Items when focused.
- Announcing when the autocomplete menu is expanded/collapsed.

### Actual Behavior

Recording of Autocomplete interaction in Jaws where:
- It is hard to tell how many items are in the results
- focused Autocomplete items are not read aloud
- expanded/collapsed state of the Autocomplete menu is not announced

https://github.com/user-attachments/assets/5e7f5140-2533-47e1-bd86-79abc3ffa2fa

### Expected Behavior

The UK Gov design systems' component](https://github.com/alphagov/accessible-autocomplete), along with [their examples](https://alphagov.github.io/accessible-autocomplete/examples/) provide a great example of this implementation. The recording is navigating the UK Gov's examples using JAWS.

https://github.com/user-attachments/assets/84ef0892-17dd-4244-8f7e-2ce0cf8b33b9

### Reproduction Sample

https://codepen.io/DintaMel-the-lessful/pen/YPNJeEz?editors=100

### Reproduction Steps

1. Open JAWS
2. Navigate to an example that uses Autocomplete.
3. Type something in the search input. I typed "the" in my example.
4. Observe that it's unclear how many items are in the results.
5. Press the down arrow to navigate to the results.
6. Observe that autocomplete-item results are not read aloud.

### Reproduction Version

5.1

### Relevant Info

This issue is being opened as a result of [ENH-000157979](https://desktopqa/SFTriage?sfid=ENH-000157979) is SF Triage. Initially changes were made to the `arcgis-search` component. We've determined the next steps for improving the `arcgis-search` component are to improve the accessibility experience of the underlying Autocomplete and Autocomplete Item components. See conversation in [ArcGIS JS API issue #51650](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/51650#issuecomment-6481643).

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript
